### PR TITLE
APIKIT-2677: Fix issue validating number parameters with leading zeros

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    assertFalse(floatUriParam.validate("000.xz"));
+    assertFalse(floatUriParam.validate("01253a"));
+
+    Parameter doubleUriParam = resources.get("/{double}").getAction(GET_ACTION).getResolvedUriParameters().get("double");
+    assertTrue(doubleUriParam.validate("00123"));
+    assertTrue(doubleUriParam.validate("00.123"));
+    assertTrue(doubleUriParam.validate("0"));
+    assertTrue(doubleUriParam.validate(String.valueOf(Double.MAX_VALUE)));
+
+    assertFalse(doubleUriParam.validate("hola"));
```
Comment (dcourtil) [inline parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java:86]: False assertions are all repeated, try adding a method parametrized with the Parameter

File: parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    assertTrue(integerUriParam.validate("0"));
+    assertTrue(integerUriParam.validate("00123"));
+
+    assertFalse(integerUriParam.validate("hola"));
+    assertFalse(integerUriParam.validate("000.xz"));
+    assertFalse(integerUriParam.validate("01253a"));
+    assertFalse(integerUriParam.validate(Long.MAX_VALUE + "1"));
+
+    Parameter floatUriParam = resources.get("/{float}").getAction(GET_ACTION).getResolvedUriParameters().get("float");
+    assertTrue(floatUriParam.validate("0"));
```
Comment (dcourtil) [inline parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java:71]: Are float and double with a leading dot allowed?
For example ".0321" or ".123"

File: interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java [hunk 3]
```diff
@@ -30,18 +31,47 @@
...
       return nullValidationReport.get();
     }
 
-    return jsonValidator.get().syncValidate(APPLICATION_JSON,  needsQuotes ?
-            quote(value.replaceAll("\"", "\\\\\"")) : value);
+    return jsonValidator.get().syncValidate(APPLICATION_JSON, getPayload(value));
+  }
+
+  private String getPayload(String value) {
+    if(needsQuotes)
```
Comment (dcourtil) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java:49]: Add curly brackets to if statements

File: interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java [hunk 4]
```diff
@@ -30,18 +31,47 @@
...
+      return quote(value.replaceAll("\"", "\\\\\""));
+
+    if(isBoolean)
+      return value;
+
+    return removeLeadingZeros(value);
+  }
+
+  private String removeLeadingZeros(String value) {
```
Comment (dcourtil) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java:58]: Add unit test for this method

Comment (dcourtil) [reply interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java:58]: nm, did't notice it is private

File: parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java (hunk 2 above)
Comment (juanchib) [reply parser-service/src/test/java/org/mule/parser/service/UriParamsTestCase.java]: Not allowed on AMF and allowed at java parser

File: interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java [hunk 5]
```diff
@@ -30,18 +31,49 @@
...
+
+    if(isBoolean) {
+      return value;
+    }
+    
+    return removeLeadingZeros(value);
+  }
+
+  private String removeLeadingZeros(String value) {
+    if(!value.startsWith("0"))
```
Comment (dcourtil) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/model/JsonParameterValidationStrategy.java:61]: Curly braces missing

**Pull request merged**